### ISSUE TITLE
#371 (overrides) adds a breaking change that does not cause runtime errors

### ISSUE THREAD

**Issue by Dimava** [antfu/eslint-config#382]
### Describe the bug

#371 causes root `overrides` property to be ignored, but does not result in any type or runtime errors, making the linting break

### Reproduction

https://github.com/Dimava/antfu-eslint-config/tree/repro/371
(basically, an empty project with 1 rule to test overrides - they don't seem to work)

### System Info

```Shell
"@antfu/eslint-config": "^2.6.1",
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
